pkg/impersonation/impersonation.go

```diff
@@ -323,3 +344,84 @@ func (i *Impersonator) waitForServiceAccount(sa *corev1.ServiceAccount) (*corev1
 	}
 	return ret, nil
 }
+
+func getUser(clusterContext *config.UserContext, userInfo user.Info, groupName string) (user.Info, error) {
```

Comment:
I am concerned with how much processing we are adding in front of every API call. This also seems like we are trying to gather info from multiple places. Where is our source of truth for the user?

Reply:
This function is the source of truth for both the role template controller and the proxy handler. The source of truth for groups is the `UserAttributes` lister. The source of truth for extras is the `User` lister plus the `user.Info` object. Combining these for extras is necessary to address this bug and to avoid regressing on https://github.com/rancher/rancher/issues/33912

Reply:
We could remove all processing and revert to what we currently have but with retries to fix the conflict errors.